numpy/ma/tests/test_deprecations.py

```diff
@@ -66,3 +66,23 @@ def test_axis_default(self):
         result = ma_max(data1d)
         assert_equal(result, ma_max(data1d, axis=None))
         assert_equal(result, ma_max(data1d, axis=0))
+
+
+class TestFromtextfile:
+    def test_fromtextfile_delimitor(self):
+        # NumPy 1.22.0, 2021-09-22
+
+        fcontent = (
+"""
+A,B,C,D
+'string 1';1;1.0;'mixed column'
+'string 2';2;2.0;
+'string 3';3;3.0;123
+'string 4';4;4.0;3.14
+""")
+        with temppath() as path:
+            with open(path, 'w') as f:
+                f.write(fcontent)
+            result = assert_warns(DeprecationWarning,
```

Comment:
I would use `pytest.warns`, or the `self.assert_deprecated` (I don't care about it here, this is python code so we don't really have to try the error case explicitly).

Reply:
Again I have tried to mimic the rest of [`numpy/ma/tests/test_deprecations.py`](https://github.com/numpy/numpy/blob/main/numpy/ma/tests/test_deprecations.py), for consistency.

Reply:
Unless you are tired of updating... we have to start somewhere to use the better versions for these things (`assert_deprecated` doesn't apply, this is a different file.)

Reply:
:smile: Will do...

Reply:
`pytest.warns` done.